enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,246 @@
+support and will not allow switching between runtime.
+
+## Containerd logging
+Containerd can be started with parameters in which we can enable logging and specify the file path to
+log warnings/errors. Log files will be stored at c:\var\log\containerd. Pod logs will be stored in the
+same location (C:\var\log\pods)
+
+## Upgrade
+WMCO supports upgrade. Please refer to the [WMCO-Upgrade](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+enhancement proposal document to learn more about WMCO upgrade.
+
+The procedure for an upgrade that includes migration to containerd is as follows:
+1) MachineSet
+   * As part of upgrade Machine object gets deleted, resulting in the drain and deletion of the Windows node.
+   * New Windows VM will be created.
+   * The upgraded WMCO instance will configure Windows VM with containerd as the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * Upgrade process will uninstall kubelet, kube-proxy, CNI, and hybrid-overlay components that were installed by WMCO.
+   * Windows OS-specific configurations like HNS-Network that were created by WMCO instance would be deleted.
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kubelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+   * If the Docker service is present, it will continue to run.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet. We work closely with the
+  containerd open source community and try to resolve it as soon as possible. We aim to fix any critical issue in the
+  upstream first and then update the containerd submodule to point to the latest containerd upstream version. If there
+  is any critical issue that needs to be addressed as soon as possible, as an exception we carry patch(es) and then
+  follow up with upstream.
+* As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release. We will use two test jobs to test all e2e test cases
+  in the 4.10 community branch(OKD) and 4.10 branch(ccm OCP)to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
```

Comment:
Link should have text:
[PR](https://github.com/containerd/containerd/pull/6150)